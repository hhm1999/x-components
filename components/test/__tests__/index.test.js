import { mount } from '@vue/test-utils'
import Btn from '../btn'

describe('Btn', () => {
  it('显示正确', () => {
    const wrapper = mount({
      render() {
        return <Btn>按钮</Btn>
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('禁用正常', () => {
    const wrapper = mount({
      data() {
        return {
          isClick: false,
        }
      },
      methods: {
        testClick() {
          this.isClick = true
        },
      },
      render() {
        return <Btn disabled onClick={this.testClick}>按钮</Btn>
      }
    })
    expect(wrapper.find('.disabled').exists()).toBe(true)
    wrapper.trigger('click')
    expect(wrapper.vm.isClick).toBe(false)
  })
})