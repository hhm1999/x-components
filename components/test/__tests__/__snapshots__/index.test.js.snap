// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Btn 显示正确 1`] = `<div class="x-btn c_noselect no_disabled main default-size default">按钮</div>`;
